File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Core/System.String/String.IsNotNullOrEmpty.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

Public Module Extensions_507

	''' <summary>
	'''     A string extension method that queries if '@this' is not (null or empty).
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <returns>true if '@this' is not (null or empty), false if not.</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function IsNotNullOrEmpty(this As String) As Boolean
		Return Not String.IsNullOrEmpty(this)
	End Function
End Module

File: src/Z.Core/System.String/String.IsNotNullOrEmpty.cs
// Description: C# Extension Methods | Enhance the .NET Framework and .NET Core with over 1000 extension methods.
// Website & Documentation: https://csharp-extension.com/
// Issues: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License (MIT): https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: https://zzzprojects.com/
// Copyright © ZZZ Projects Inc. All rights reserved.
public static partial class Extensions
{
    /// <summary>
    ///     A string extension method that queries if '@this' is not (null or empty).
    /// </summary>
    /// <param name="this">The @this to act on.</param>
    /// <returns>true if '@this' is not (null or empty), false if not.</returns>
    public static bool IsNotNullOrEmpty(this string @this)
    {
        return !string.IsNullOrEmpty(@this);
    }
}